tracker/tracker/src/main/modules/mouse.ts

```diff
@@ -120,48 +120,58 @@ export default function (app: App): void {
     }
   }
 
-  const selectorMap: { [id: number]: string } = {}
-  function getSelector(id: number, target: Element): string {
-    return (selectorMap[id] = selectorMap[id] || _getSelector(target))
+  const patchDocument = (document: Document) => {
+    const selectorMap: { [id: number]: string } = {}
+    function getSelector(id: number, target: Element): string {
+      return (selectorMap[id] = selectorMap[id] || _getSelector(target, document))
+    }
+
+    app.attachEventListener(document.documentElement, 'mouseover', (e: MouseEvent): void => {
+      const target = getTarget(e.target, document)
+      if (target !== mouseTarget) {
+        mouseTarget = target
+        mouseTargetTime = performance.now()
+      }
+    })
+    app.attachEventListener(
+      document,
+      'mousemove',
+      (e: MouseEvent): void => {
+        const { top, left } = app.observer.getDocumentOffset(document)
+        mousePositionX = e.clientX + left
+        mousePositionY = e.clientY + top
+        mousePositionChanged = true
+      },
+      false,
+    )
+    app.attachEventListener(document, 'click', (e: MouseEvent): void => {
+      const target = getTarget(e.target, document)
+      if ((!e.clientX && !e.clientY) || target === null) {
+        return
+      }
+      const id = app.nodes.getID(target)
+      if (id !== undefined) {
+        sendMouseMove()
+        app.send(
+          MouseClick(
+            id,
+            mouseTarget === target ? Math.round(performance.now() - mouseTargetTime) : 0,
+            getTargetLabel(target),
+            getSelector(id, target),
+          ),
+          true,
+        )
+      }
+      mouseTarget = null
+    })
   }
 
-  app.attachEventListener(document.documentElement, 'mouseover', (e: MouseEvent): void => {
-    const target = getTarget(e.target)
-    if (target !== mouseTarget) {
-      mouseTarget = target
-      mouseTargetTime = performance.now()
+  app.nodes.attachNodeCallback((node) => {
+    if (isDocument(node)) {
+      patchDocument(node)
     }
   })
-  app.attachEventListener(
-    document,
-    'mousemove',
-    (e: MouseEvent): void => {
-      mousePositionX = e.clientX
-      mousePositionY = e.clientY
-      mousePositionChanged = true
-    },
-    false,
-  )
-  app.attachEventListener(document, 'click', (e: MouseEvent): void => {
-    const target = getTarget(e.target)
-    if ((!e.clientX && !e.clientY) || target === null) {
-      return
-    }
-    const id = app.nodes.getID(target)
-    if (id !== undefined) {
-      sendMouseMove()
-      app.send(
-        MouseClick(
-          id,
-          mouseTarget === target ? Math.round(performance.now() - mouseTargetTime) : 0,
-          getTargetLabel(target),
-          getSelector(id, target),
-        ),
-        true,
-      )
-    }
-    mouseTarget = null
-  })
+  patchDocument(document)
 
   app.ticker.attach(sendMouseMove, 10)
```

Comment:
and connect this const with this one as well maybe

Reply:
This ticker arg determines sending frequency. Why do we need another constant that describes the same thing?

Or do you mean to add a configurable option?

Reply:
yes

Reply:
yep, good idea. not here though